### DISCUSSION TITLE
it is possible to use pc as a server?

### DISCUSSION THREAD

**Discussion by ahmeedev** [serverpod/serverpod#1059] [Q&A] [answered]
I want to deploy my API on my 2nd pc with a list of API endpoints. and, I want to make it accessible to everyone publically. Is it possible to change the API and database's IP addresses so that everyone can access API completely?

**Comment by Isakdl** [comment, marked as answer]
Yes it is possible, but not necessarily trivial or advisable. The configurations you need to make is mostly outside of Serverpod and you need some understanding of networking, firewalls and security configurations to set this up. 

I _recommend_ you to run your publicly hosted server on a cloud provider. There are premade Terraform deployment scripts for serverpod that setups the entire infrastructure for you, take a look at the documentation [here](https://docs.serverpod.dev/deployments/deployment-strategy). If you run on GCP you can use the cloud engine setup, and Google does provide [$300 of credits](https://cloud.google.com/free) for new customers meaning you can do this for free for quite some time!

By hosting your API externally, you can focus on developing and maintaining your application without worrying about the intricacies of network configuration and potential security risks associated with exposing your infrastructure to the public internet.

---

In Serverpod, there is a configuration file `config/production.yaml` in the server project where you can specify the expected public domain. This setting primarily affects the URLs generated by Serverpod to correctly point to resources such as images. However, it does not directly handle making your API accessible to the public. The rest of the needed configurations are hard to help you with as it depends heavily on your hardware, network configuration and your ISP and are done outside of the Serverpod framework.

Before attempting to host the API yourself, it is important to note that it is generally not recommended for several reasons:

* Dynamic IP addresses: Most residential or consumer-grade internet connections do not have static IP addresses. This means that your IP address could change periodically, resulting in frequent disruptions to your API's accessibility.

* Firewall and router configurations: Hosting your API at home would require you to open specific ports in your firewall and router settings to allow incoming traffic. This can be complex and potentially introduce security vulnerabilities if not properly configured.

* Network redirection: You would need to redirect incoming traffic from your router to your internal network where the API is hosted. Misconfiguration in this process can lead to improper routing or even exposing your entire network to the open internet, posing a significant security risk.

* Domain usage: It is highly recommended to use a domain name and point it to your API's IP address. However, this can be problematic if your IP address is not static, as mentioned earlier. Dynamic DNS services can help mitigate this issue, but they introduce additional complexity.

* Database security: Exposing your database on a public IP address is extremely risky and not recommended. Databases typically contain sensitive information, and direct public access can lead to unauthorized access or data breaches. It is crucial to ensure that your database is not accessible from the public internet.

Utilizing a dedicated hosting provider or cloud service can offer a more secure and scalable solution. These services typically provide robust security measures, static IP addresses, and simplified domain management.
